util/model_initializer.py

```diff
@@ -80,11 +95,7 @@ def _create_path(self, matrix_name, matrix_shape, config=None):
         if 'n_iterations' not in config.keys():
             config['n_iterations'] = self.config['n_iterations']
         config['n_factors'] = matrix_shape[1]
-        generated_key = ''
         config['n_rows'] = n_rows
-        keys_array = sorted(config)
-        for key in keys_array:
-            generated_key += key + ':'
-            generated_key += str(config[key]) + ','
-        path = generated_key.strip(',') + matrix_name
-        return os.path.join(os.path.dirname(os.path.realpath(__file__)), '../', self.folder, path + '.dat')
+        path = self._generate_path(config, matrix_name)
+        base_dir = os.path.dirname(os.path.realpath(__file__))
+        return os.path.join(os.path.dirname(base_dir), self.folder, path + '.dat')
```

Comment:
They are separated because if we want to change the file name convention later on (which I did a small bit) , it would be well structured, because I think later on we can use shorter names, so it would be easier to change then. and I also see it's well structured.